libiocage/lib/Jail.py

```diff
@@ -18,6 +18,49 @@
 
 
 class Jail:
+    """
+    Iocage unit orchestrates a jail's configuration and manages state
+
+    Jails are represented as zfs dataset `zpool/iocage/jails/<NAME>`
+
+    Directory Structure:
+
+        zpool/iocage/jails/<NAME>: The jail's dataset containing it's
+            configuration and root directory. Iocage legacy used to store
+            a jails configuration as ZFS properties on this dataset. Even
+            though the modern JSON config mechanism is preferred.
+
+        zpool/iocage/jails/<NAME>/root: This directory is the dataset
+            used as jail path when starting a jail. Usually the origin
+            or a root dataset is a release root directory
+
+        zpool/iocage/jails/<NAME>/config.json: Jails configured with the latest
+            configuration style store their information in a JSON file. When
+            this file is found in the jail's dataset, libiocage assumes
+            the jail to be a JSON-style jail and ignores other configuration
+            mechanisms
+
+        zpool/iocage/jails/<NAME>/config: Another supported configuration
```

Comment:
Drop the word supported, it isn't supported, but is able to be read. We need to offer a cutoff, we don't want to juggle supporting 3 different configuration methods.